ui/App/SettingsScreen.svelte

```diff
@@ -95,6 +118,40 @@
   ];
 
   const session = Session.unsealed();
+
+  let colorHex: string = "";
+
+  const hexToRgb = (hex: string) => {
+    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return result
+      ? {
+          r: parseInt(result[1], 16),
+          g: parseInt(result[2], 16),
+          b: parseInt(result[3], 16),
+        }
+      : null;
+  };
+
+  $: if ($primaryColorStore === "hex") {
+    const colorRgb = hexToRgb(colorHex);
+    const root = document.documentElement;
+    root.style.setProperty(
+      "--color-primary",
+      `rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},1`
+    );
+    root.style.setProperty(
+      "--color-primary-level-1",
+      `rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},0.17`
+    );
+    root.style.setProperty(
+      "--color-primary-level-2",
+      `rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},0.37`
+    );
+    root.style.setProperty(
+      "--color-primary-level-6",
+      `rgba(${colorRgb.r},${colorRgb.g},${colorRgb.b},0.87`
+    );
+  }
```

Comment:
@rudolfs if here I could target `[data-primarycolor="hex"]` instead of `root` it would work imo

Reply:
I’ll take a look at this. Maybe we can do something with the StyleSheet DOM APIs

Reply:
This should be possible with the [CSSStyleSheet](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule) API.